@nx/dependency-checks does not seem to support pnpm-catalogs

### Current Behavior

While using `pnpm-catalogs` i get error's on dependency checks that should pass.

https://pnpm.io/catalogs

### Expected Behavior

Would expect pnpm-catalogs to work with the dependency-check lint rule.

### GitHub Repo

https://github.com/ryanbas21/epb-monorepo/tree/update-app-settings

### Steps to Reproduce

1. pnpm install
2. pnpm nx affected -t lint




### Nx Report

```shell
NX   Report complete - copy this into the issue template

Node           : 23.6.0
OS             : darwin-arm64
Native Target  : aarch64-macos
pnpm           : 9.15.0

nx                 : 20.4.0
@nx/js             : 20.4.0
@nx/jest           : 20.4.0
@nx/eslint         : 20.4.0
@nx/workspace      : 20.4.0
@nx/devkit         : 20.4.0
@nx/esbuild        : 20.4.0
@nx/eslint-plugin  : 20.4.0
@nx/node           : 20.4.0
@nx/vite           : 20.4.0
@nx/web            : 20.4.0
typescript         : 5.7.3
---------------------------------------
Registered Plugins:
@nx/js/typescript
@nx/vite/plugin
@nx/eslint/plugin
```

### Failure Logs

```
        19:5  error  The version specifier does not contain the installed version of "@effect/platform" package: 0.75.3  @nx/dependency-checks
        20:5  error  The version specifier does not contain the installed version of "effect" package: 3.12.9
```

### Package Manager Version

pnpm 9.15
### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

snippet from pnpm-lock file
```
lockfileVersion: '9.0'

settings:
  autoInstallPeers: true
  excludeLinksFromLockfile: false

catalogs:
  effect:
    '@effect/platform':
      specifier: ^0.75.3
      version: 0.75.3
    '@effect/platform-node':
      specifier: ^0.71.3
      version: 0.71.3
    '@effect/vitest':
      specifier: ^0.17.5
      version: 0.17.5
    effect:
      specifier: ^3.12.9
      version: 3.12.9

importers:

  .:
    dependencies:
      lint-staged:
        specifier: ^15.4.3
        version: 15.4.3
    devDependencies:
      '@effect/vitest':
        specifier: catalog:effect

```

My guess is that the specifier here being `catalog:effect` is causing issues.

If I'm missing something though let me know.

--- a/packages/eslint-plugin/src/rules/dependency-checks.spec.ts
+++ b/packages/eslint-plugin/src/rules/dependency-checks.spec.ts
@@ -1637,6 +1637,118 @@ describe('Dependency checks (eslint)', () => {
     );
     expect(failures.length).toEqual(0);
   });
+  it('should not report workspace:^ and workspace:~', () => {
+    const packageJson = {
+      name: '@mycompany/liba',
+      dependencies: {
+        external1: 'workspace:~',
+        external2: 'workspace:^',
+      },
+    };
+
+    const fileSys = {
+      './libs/liba/package.json': JSON.stringify(packageJson, null, 2),
+      './libs/liba/src/index.ts': '',
+      './package.json': JSON.stringify(rootPackageJson, null, 2),
+    };
+    vol.fromJSON(fileSys, '/root');
+
+    const failures = runRule(
+      {},
+      `/root/libs/liba/package.json`,
+      JSON.stringify(packageJson, null, 2),
+      {
+        nodes: {
+          liba: {
+            name: 'liba',
+            type: 'lib',
+            data: {
+              root: 'libs/liba',
+              targets: {
+                build: {},
+              },
+            },
+          },
+        },
+        externalNodes,
+        dependencies: {
+          liba: [
+            { source: 'liba', target: 'npm:external1', type: 'static' },
+            { source: 'liba', target: 'npm:external2', type: 'static' },
+          ],
+        },
+      },
+      {
+        liba: [
+          createFile(`libs/liba/src/main.ts`, [
+            'npm:external1',
+            'npm:external2',
+          ]),
+          createFile(`libs/liba/package.json`, [
+            'npm:external1',
+            'npm:external2',
+          ]),
+        ],
+      }
+    );
+    expect(failures.length).toEqual(0);
+  });
+  it('should not report catalog:', () => {
+    const packageJson = {
+      name: '@mycompany/liba',
+      dependencies: {
+        external1: 'catalog:',
+        external2: 'catalog:nx',
+      },
+    };
+
+    const fileSys = {
+      './libs/liba/package.json': JSON.stringify(packageJson, null, 2),
+      './libs/liba/src/index.ts': '',
+      './package.json': JSON.stringify(rootPackageJson, null, 2),
+    };
+    vol.fromJSON(fileSys, '/root');
+
+    const failures = runRule(
+      {},
+      `/root/libs/liba/package.json`,
+      JSON.stringify(packageJson, null, 2),
+      {
+        nodes: {
+          liba: {
+            name: 'liba',
+            type: 'lib',
+            data: {
+              root: 'libs/liba',
+              targets: {
+                build: {},
+              },
+            },
+          },
+        },
+        externalNodes,
+        dependencies: {
+          liba: [
+            { source: 'liba', target: 'npm:external1', type: 'static' },
+            { source: 'liba', target: 'npm:external2', type: 'static' },
+          ],
+        },
+      },
+      {
+        liba: [
+          createFile(`libs/liba/src/main.ts`, [
+            'npm:external1',
+            'npm:external2',
+          ]),
+          createFile(`libs/liba/package.json`, [
+            'npm:external1',
+            'npm:external2',
+          ]),
+        ],
+      }
+    );
+    expect(failures.length).toEqual(0);
+  });
 
   it('should require swc if @nx/js:swc executor', () => {
     const packageJson = {

--- a/packages/eslint-plugin/src/rules/dependency-checks.spec.ts
+++ b/packages/eslint-plugin/src/rules/dependency-checks.spec.ts
@@ -30,6 +30,8 @@ const rootPackageJson = {
   dependencies: {
     external1: '~16.1.2',
     external2: '^5.2.0',
+    external3: '1.0.0',
+    external4: '1.0.0',
   },
   devDependencies: {
     tslib: '^2.1.0',
@@ -51,6 +53,22 @@ const externalNodes: Record<string, ProjectGraphExternalNode> = {
     data: {
       packageName: 'external2',
       version: '5.5.6',
+    },
+  },
+  'npm:external3': {
+    name: 'npm:external3',
+    type: 'npm',
+    data: {
+      packageName: 'external3',
+      version: '1.0.0',
+    },
+  },
+  'npm:external4': {
+    name: 'npm:external3',
+    type: 'npm',
+    data: {
+      packageName: 'external4',
+      version: '1.0.0',
     },
   },
   'npm:random-external': {
@@ -1581,12 +1599,11 @@ describe('Dependency checks (eslint)', () => {
     `);
   });
 
-  it('should not report * and workspace:*', () => {
+  it('should not report *', () => {
     const packageJson = {
       name: '@mycompany/liba',
       dependencies: {
         external1: '*',
-        external2: 'workspace:*',
       },
     };
 
@@ -1616,33 +1633,24 @@ describe('Dependency checks (eslint)', () => {
         },
         externalNodes,
         dependencies: {
-          liba: [
-            { source: 'liba', target: 'npm:external1', type: 'static' },
-            { source: 'liba', target: 'npm:external2', type: 'static' },
-          ],
+          liba: [{ source: 'liba', target: 'npm:external1', type: 'static' }],
         },
       },
       {
-        liba: [
-          createFile(`libs/liba/src/main.ts`, [
-            'npm:external1',
-            'npm:external2',
-          ]),
-          createFile(`libs/liba/package.json`, [
-            'npm:external1',
-            'npm:external2',
-          ]),
-        ],
+        liba: [createFile(`libs/liba/src/main.ts`, ['npm:external1'])],
       }
     );
     expect(failures.length).toEqual(0);
   });
-  it('should not report workspace:^ and workspace:~', () => {
+
+  it('should not report workspace: protocol', () => {
     const packageJson = {
       name: '@mycompany/liba',
       dependencies: {
         external1: 'workspace:~',
         external2: 'workspace:^',
+        external3: 'workspace:',
+        external4: 'workspace:../external4',
       },
     };
 
@@ -1675,6 +1683,8 @@ describe('Dependency checks (eslint)', () => {
           liba: [
             { source: 'liba', target: 'npm:external1', type: 'static' },
             { source: 'liba', target: 'npm:external2', type: 'static' },
+            { source: 'liba', target: 'npm:external3', type: 'static' },
+            { source: 'liba', target: 'npm:external4', type: 'static' },
           ],
         },
       },
@@ -1683,17 +1693,22 @@ describe('Dependency checks (eslint)', () => {
           createFile(`libs/liba/src/main.ts`, [
             'npm:external1',
             'npm:external2',
+            'npm:external3',
+            'npm:external4',
           ]),
           createFile(`libs/liba/package.json`, [
             'npm:external1',
             'npm:external2',
+            'npm:external3',
+            'npm:external4',
           ]),
         ],
       }
     );
     expect(failures.length).toEqual(0);
   });
-  it('should not report catalog:', () => {
+
+  it('should not report catalog: protocol', () => {
     const packageJson = {
       name: '@mycompany/liba',
       dependencies: {

--- a/packages/eslint-plugin/src/rules/dependency-checks.ts
+++ b/packages/eslint-plugin/src/rules/dependency-checks.ts
@@ -219,6 +219,14 @@ export default ESLintUtils.RuleCreator(
         npmDependencies[packageName] === '*' ||
         packageRange === '*' ||
         packageRange === 'workspace:*' ||
+        packageRange === 'workspace:~' ||
+        packageRange === 'workspace:^' ||
+        /**
+         * Catalogs can be named, or left unnamed
+         * So just checking up until the : will catch both cases
+         * e.g. catalog:some-catalog or catalog:
+         */
+        packageRange.startsWith('catalog:') ||
         satisfies(npmDependencies[packageName], packageRange, {
           includePrerelease: true,
         })

--- a/packages/eslint-plugin/src/rules/dependency-checks.ts
+++ b/packages/eslint-plugin/src/rules/dependency-checks.ts
@@ -218,9 +218,7 @@ export default ESLintUtils.RuleCreator(
         packageRange.startsWith('file:') ||
         npmDependencies[packageName] === '*' ||
         packageRange === '*' ||
-        packageRange === 'workspace:*' ||
-        packageRange === 'workspace:~' ||
-        packageRange === 'workspace:^' ||
+        packageRange.startsWith('workspace:') ||
         /**
          * Catalogs can be named, or left unnamed
          * So just checking up until the : will catch both cases

